src/dev/flang/fuir/analysis/dfa/DFA.java

```diff
@@ -1849,6 +1864,101 @@ static void setArrayElementsToAnything(Call cl, int argnum, String intrinsicName
         jobj.setField(cl._dfa, jref, Value.UNKNOWN_JAVA_REF);
         return jobj;
       });
+  }
+  static Value newFuzionJavaCall(Call cl) {
+    var rc = cl._dfa._fuir.clazzResultClazz(cl._cc);
+    if (cl._dfa._fuir.clazzBaseName(rc).startsWith("outcome"))
+      {
+        var oc = cl._dfa._fuir.clazzChoice(rc, 0);
+        return switch (cl._dfa._fuir.getSpecialId(oc))
+          {
+            case c_i8, c_u16, c_i16, c_i32, c_i64,
+              c_f32, c_f64, c_bool, c_unit -> {
+              var v = switch (cl._dfa._fuir.getSpecialId(oc))
+                {
+                  case c_i8, c_u16, c_i16, c_i32, c_i64,
+                    c_f32, c_f64 -> new NumericValue(cl._dfa, oc);
+                  case c_bool -> cl._dfa._bool;
+                  case c_unit -> Value.UNIT;
+                  default -> Value.UNIT;
+                };
+              var okay = new TaggedValue(cl._dfa, oc, v, 0);
+              var err = new TaggedValue(cl._dfa, oc, cl._dfa.newInstance(cl._dfa._fuir.clazzChoice(rc, 1), null), 1);
+              yield okay.join(err);
+            }
+            default -> {
+              var jref = cl._dfa._fuir.lookupJavaRef(oc);
+              var jobj = cl._dfa.newInstance(oc, null);
+              jobj.setField(cl._dfa, jref, Value.UNKNOWN_JAVA_REF);
+              var okay = new TaggedValue(cl._dfa, rc, jobj, 0);
+              var err = new TaggedValue(cl._dfa, rc, cl._dfa.newInstance(cl._dfa._fuir.clazzChoice(rc, 1), null), 1);
+              yield okay.join(err);
```

Comment:
These three lines are repeated compared to the previous case (except for the likely bug that `oc` is used there, and the different variable name `v` vs. `jobj`). Can you try to move these out of the switch and avoid the duplication?